### PR TITLE
Refactor Migrations Implementation

### PR DESCRIPTION
**Pull request by aryanmehrotra**
## Pull Request Template
1. Remove unnecessary lines of code.
2. Update workflow to keep migrations part of pkg testing

### CODE REVIEW

File: pkg/gofr/migration/interface.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	goRedis "github.com/redis/go-redis/v9"
+)
+
+type Decorator interface {
```
Comment (vipul-rawat) [inline pkg/gofr/migration/interface.go:11]: This could be named a bit meaningful!

**Review by vipul-rawat — CHANGES_REQUESTED** [review]

File: pkg/gofr/migration/datasource.go [hunk 2]
```diff
@@ -1,60 +1,10 @@
...
-	"time"
-
-	goRedis "github.com/redis/go-redis/v9"
-
-	"gofr.dev/pkg/gofr/container"
-	gofrSql "gofr.dev/pkg/gofr/datasource/sql"
-)
-
 type Datasource struct {
+	// TODO this should not be embedded rather it should be
```
Comment (Umang01-hash) [inline pkg/gofr/migration/datasource.go:4]: I think it's an incomplete TODO left, please complete it

File: pkg/gofr/migration/manager_test.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+	"github.com/stretchr/testify/assert"
+
+	"gofr.dev/pkg/gofr/container"
+	"gofr.dev/pkg/gofr/testutil"
+)
+
+func Test_getMigratorDatastoreNotInitialised(t *testing.T) {
+	logs := testutil.StdoutOutputForFunc(func() {
+		container, _ := container.NewMockContainer(t)
```
Comment (Umang01-hash) [inline pkg/gofr/migration/manager_test.go:14]: please name it to `mockContainer` as name **container** is colliding with imported package name

File: pkg/gofr/migration/interface.go (hunk 1 above)
Comment (aryanmehrotra) [reply pkg/gofr/migration/interface.go]: Have removed the interface! as it was not being used externally!

**Review by Umang01-hash — CHANGES_REQUESTED** [review]

**Review by vipul-rawat — APPROVED** [review]

**Review by srijan-27 — APPROVED** [review]

**Review by Umang01-hash — APPROVED** [review]

**Pull request merged**